File: app/src/main/java/com/soya/launcher/ui/fragment/FocusFragment.java
package com.soya.launcher.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.soya.launcher.R;

public class FocusFragment extends AbsFragment implements View.OnClickListener {

    public static FocusFragment newInstance() {
        
        Bundle args = new Bundle();
        
        FocusFragment fragment = new FocusFragment();
        fragment.setArguments(args);
        return fragment;
    }

    private View mNextView;

    @Override
    public int getLayoutId() {
        return R.layout.fragment_focus;
    }

    @Override
    protected void init(View view, LayoutInflater inflater) {
        super.init(view, inflater);
        mNextView = view.findViewById(R.id.next);
    }

    @Override
    protected void initBind(View view, LayoutInflater inflater) {
        super.initBind(view, inflater);
        mNextView.setOnClickListener(this::onClick);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        requestFocus(mNextView);
    }

    @Override
    public void onClick(View v) {
        if (v.equals(mNextView)){
            getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.main_browse_fragment, GuideGroupGradientFragment.newInstance()).addToBackStack(null).commit();
        }
    }

    @Override
    protected int getWallpaperView() {
        return R.id.wallpaper;
    }
}

File: app/src/main/java/com/soya/launcher/ui/fragment/FocusFragment.kt
package com.soya.launcher.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import androidx.fragment.app.FragmentManager
import com.shudong.lib_base.base.BaseViewModel
import com.shudong.lib_base.ext.e
import com.shudong.lib_base.ext.obseverLiveEvent
import com.shudong.lib_base.ext.stringValue
import com.soya.launcher.BaseWallPaperFragment
import com.soya.launcher.R
import com.soya.launcher.cache.AppCache
import com.soya.launcher.databinding.FragmentFocusBinding
import com.soya.launcher.ext.navigateTo

class FocusFragment : BaseWallPaperFragment<FragmentFocusBinding, BaseViewModel>(),
    View.OnClickListener {


    override fun initView() {
        mBind.tvNext.setOnClickListener { v: View -> this.onClick(v) }
        mBind.tvNext.apply { post { requestFocus() } }

    }


    override fun onClick(v: View) {
        if (v == mBind.tvNext) {
            requireActivity().supportFragmentManager.navigateTo(
                R.id.main_browse_fragment,
                GuideLanguageFragment.newInstance(), tag = GuideLanguageFragment::class.simpleName
            )
        }
    }

    override fun excuteLang() {
        mBind.tvTip.text = R.string.focus_guide.stringValue()
        mBind.tvNext.text = R.string.next.stringValue()

    }



    companion object {
        @JvmStatic
        fun newInstance(): FocusFragment {
            val args = Bundle()

            val fragment = FocusFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
